features/google/src/model/importers/activities-importer-html.js

```diff
@@ -12,15 +12,97 @@ class ActivityHtmlParser {
     _extractDate(text) {
         // Ignoring the timestamp for now - we don't need hourly accuracy at the
         // time of writing this.
-        const datePattern =
-            /([A-Z][a-z][a-z]) ([0-9]{1,2}), ([0-9]{4,4}), ([0-9:]{7,8} [A|P]M) ([A-Z]{3,4})/;
-        const match = text.match(datePattern);
-        const month = match?.[1];
-        const day = match?.[2];
-        const year = match?.[3];
-
-        const dateString = `${day} ${month} ${year}`;
-        return dateString ? new Date(dateString) : null;
+
+        // Known formats:
+        // 3 Feb 2001, 16:05:06 UTC
+        // Feb 3, 2001, 4:05:06 PM UTC
+        // 03.02.2001, 16:05:06 UTC
+
+        let date = new Date(text);
+
+        if (date && date !== null && date.toString() !== "Invalid Date") {
```

Comment:
```suggestion
        if (date && date.toString() !== "Invalid Date") {
```
second null check seems redundant